plugins/scheduled_task/lib/MediaRepurposingHandler.php

```diff
@@ -70,19 +72,29 @@ public function shouldConsumeDataChangedEvent(BaseObject $object, $previousVersi
 
 	private function getMRPWithMetadataSearchByProfile($partnerId, $metadataProfileId)
 	{
-		$allMediaRepurposingProfilesOnPartner = ScheduledTaskProfilePeer::retrieveBySystemName(MediaRepurposingUtils::MEDIA_REPURPOSING_SYSTEM_NAME, $partnerId);
-		$mediaRepurposingProfilesWithSearchOnGivenMetadataId = array();
-		foreach($allMediaRepurposingProfilesOnPartner as $mediaRepurposingProfile) {
-			/* @var $mediaRepurposingProfile ScheduledTaskProfile*/
-			if (!$mediaRepurposingProfile->getObjectFilter() || !$mediaRepurposingProfile->getObjectFilter()->getAdvancedSearch())
-				continue;
-			$items = $mediaRepurposingProfile->getObjectFilter()->getAdvancedSearch()->getItems(); // always have items in advance search because that how the MR is build
-			foreach($items as $item) {
-				if ($item instanceof MetadataSearchFilter && $item->getMetadataProfileId() == $metadataProfileId)
-					$mediaRepurposingProfilesWithSearchOnGivenMetadataId[] = $mediaRepurposingProfile->getId();
-			}
-		}
-		return $mediaRepurposingProfilesWithSearchOnGivenMetadataId;
+        	$allMediaRepurposingProfilesOnPartner = ScheduledTaskProfilePeer::retrieveBySystemName(MediaRepurposingUtils::MEDIA_REPURPOSING_SYSTEM_NAME, $partnerId);
+        	$mediaRepurposingProfilesWithSearchOnGivenMetadataId = array();
+        	$ignoreMRPResetOnMetadataUpdateConf = kConf::get(self::IGNORE_MRP_RESET_ON_METADATA_UPDATE, kConfMapNames::RUNTIME_CONFIG, array());

```

Comment:
adjust with new name and then you wont need to init the array or explode

Reply:
done